std/random.d

```diff
@@ -1524,6 +1518,381 @@ version (AnyARC4Random)
+        ulong gamma = atomicLoad!(MemoryOrder.raw)(_globalGamma);
+        if (!gamma)
+        {
+            synchronized
+            {
+                gamma = atomicLoad(_globalGamma);
+                if (!gamma)
+                {
+                    // gamma hasn't been set yet, so initialize.
+                    ulong initialSeed = void;
+                    oneTimeInitSeedAndGamma(initialSeed, gamma);
+                    atomicStore(_globalSeed, initialSeed);
+                    atomicStore(_globalGamma, gamma);
+                }
+            }
+            assert(gamma);
+        }
+        // Stafford Mix05
+        ulong x = atomicOp!"+="(_globalSeed, gamma);
+        x = (x ^ (x >>> 31)) * 0x79c1_35c1_674b_9addUL;
+        x = (x ^ (x >>> 29)) * 0x54c7_7c86_f691_3e45UL;
+        return x ^ (x >>> 30);
+    }
+    /+
+    Similar to above but without 64-bit atomic ops. (All current platforms
+    supported by DMD or LDC have 64-bit atomic ops, but in the future some
+    might not.)
+    +/
+    static if (!has64BitCAS)
+    private ulong globalSplitMix()()
+    {
+        import core.atomic : atomicLoad, atomicOp, atomicStore, MemoryOrder;
+        shared static uint _globalSeedLow;
+        shared static uint _globalSeedHigh;
+        shared static uint _globalGammaLow;
+        shared static uint _globalGammaHigh;
+        uint gammaLow = atomicLoad!(MemoryOrder.raw)(_globalGammaLow);
+        if (!gammaLow)
+        {
+            synchronized
+            {
+                gammaLow = atomicLoad(_globalGammaLow);
+                if (!gammaLow)
+                {
+                    // gamma hasn't been set yet, so initialize.
+                    ulong initialSeed = void, gamma = void;
+                    oneTimeInitSeedAndGamma(initialSeed, gamma);
+                    atomicStore(_globalSeedHigh, cast(uint) (initialSeed >>> 32));
+                    atomicStore(_globalSeedLow, cast(uint) initialSeed);
+                    gammaLow = cast(uint) gamma;
+                    uint h = cast(uint) (gamma >>> 32);
+                    assert(h);
+                    atomicStore(_globalGammaHigh, h);
+                    atomicStore(_globalGammaLow, gammaLow);
+                }
+            }
+            assert(gammaLow);
+        }
+        uint gammaHigh = atomicLoad!(MemoryOrder.raw)(_globalGammaHigh);
+        if (!gammaHigh)
+        {
+             gammaHigh = atomicLoad(_globalGammaHigh);
+             assert(gammaHigh);
+        }
+        // Under contention updates can be interleaved. This is completely fine.
+        uint low = atomicOp!"+="(_globalSeedLow, gammaLow);
+        uint high = atomicOp!"+="(_globalSeedHigh, gammaHigh + uint(low < gammaLow));
+        // Stafford Mix04
+        ulong x = (cast(ulong) high) << 32 + cast(ulong) low;
+        x = (x ^ (x >>> 33)) * 0x62a9_d9ed_7997_05f5UL;
+        x = (x ^ (x >>> 28)) * 0xcb24_d0a5_c88c_35b3UL;
+        return x ^ (x >>> 32);
```

Comment:
Have I mixed things up, or is this a different implementation of the state => variate transformation than in previous iterations of this code?

If so, why the difference?

Reply:
It is different to make it obvious that this function is not required or intended to be algorithmically identical to the other function, to make [confusion regarding intent](https://github.com/dlang/phobos/pull/6388#pullrequestreview-109545218) less likely.